src/adr/0022-cassandra-message-inconsistency.md

```diff
@@ -54,8 +54,10 @@ The message is UNSEEN in IMAP
 
 ## Decision
 
-Adopt `imapUidTable` as a source of truth. All writes will be performed to `messageIdTable` then performed on
-`imapUidTable` if successful.
+Adopt `imapUidTable` as a source of truth. Because `messageId` allows tracking changes to messages accross mailboxes 
```

Comment:
you mean Adopt `messageIdTable` as a source of truth?

Reply:
No. `imapUidTable` is indexed by message id.

The corresponding java name is less misleading: MessageIdToImapUid

Here is the corresponding comment: Holds mailbox and flags for each message, lookup by message ID

Sounds like what you want...